Upgrade project to Liferay 7.1

kriegaex opened the pull request:
**Attention!** This PR also encompasses the yet unmerged PR #24 - sorry for that, but I do not want to wait until that one was also reviewed. The new commits start from today (2018-10-18), i.e. at the time of writing there are three (3) commits specific to this branch.

What was done is mentioned in the commit messages, please read them. Basically it was as follows:
* Upgrade download to Liferay 7.1 + Tomcat 9.0.6
* Upgrade tests to run on LR 7.1
* In 7.0 only one module was missing in order to enable JMX hot-deployment. In 7.1 five (5) modules are needed. Thus I decided to copy them to _osgi/modules_ directly instead of trying to install them from Java, which would be impossible in the first place because `_installBundle(_getMavenDependencyPath(_HOT_DEPLOY_JMX_LISTENER_MVN))` would not work anyway before installing the server-side dependencies to make a call to `_installBundle` possible in the first place.

P.S.: This PR is an indirect result of my troubles related to #19. See also my comments there in order to understand why it is now more complicated to activate JMX hot-deployment in 7.1.

**Update:** Commit fb11446 should fix the problem with module auto-activation mentioned in the troubleshooting section of the read-me file. I still have not removed that section because maybe the problem can still occur under certain circumstances. I can only test on my machine.

**Update 2:** My assumption for the Travis CI build failure is that the initial Liferay start after download + unzip just takes too long, even longer than the 350 s configured in Maven. On my machine it takes twice as long during the first time due to validation: `org.apache.catalina.startup.Catalina.start Server startup in 692020 ms`

Commit:
Fix bogus Windows CMD syntax for setting environment variables

Commit:
Remove unnecessary double quotes for setting environment variables on Windows

Commit:
Make sure 'arquillian.liferay.bundle' is an absolute path (helps run the project on Windows)

For this we use 'maven.multiModuleProjectDirectory' and thus have to require at least
Maven 3.3.9 to build the project.

Commit:
Fix: A Windows path like 'c:\something' was converted to an invalid URL

Path 'C:\something' should become 'file:///C:/something' (adding another slash
in front of the drive letter), not 'file://C:/something'. The test can never work on Windows
otherwise.

Commit:
Fix: Trying to stop a not running Tomcat interrupted the build process on Windows

Thus we have to ignore a return value of 1 after trying to stop the server. For this we
need to upgrade exec-maven-plugin to 1.6.0 in order to use option 'successCode'.

Furthermore, Tomcat under Windows needs run asynchronously in order for the
build to continue and actually execute the test. For this we use ''async=true'.

Commit:
Windows fix: 'liferay.home.unix' must not contain backslashes

Create special property 'liferay.home.unix' for resource filtering in file
portal-setup-wizard.properties via build-helper-maven-plugin because Liferay
does not start correctly on Windows if the path contains backslashes.

Commit:
Add forgotten dummy file .mvn directory in order to make 'maven.multiModuleProjectDirectory' work

See https://stackoverflow.com/a/49528226/1082681.

Commit:
Extend sleep to 350 seconds when waiting for Tomcat start

This is ugly!

TODO instead: find out how to wait until server has actually started up completely
as 100 seconds could be way too short (on my machine it is more like 330 s)
or maybe too long. I am leaving this to the project mainteiners.

Commit:
Cosmetics: Add some more diagnostic output for Maven build

Commit:
Windows fix: only URL path should be prepended by '/' for drive letter

Otherwise installing JMX hot deploy listener fails.

Commit:
Upgrade from Liferay 7.0 to 7.1, adjust tests

Commit:
Small cosmetic improvements

- Liferay auto-formatter was complaining
- Sleep during Liferay start-up is not based on Maven property 'sleep.seconds',
  i.e. it can easily be changed or overridden from the command line
- One typo was fixed

Commit:
Copy JMX-related JARs to 'osgi/modules', activate Liferay developer mode

- Maven dependencies plugin now directly copies missing JMX-related modules to
  'osgi/modules' after unzipping Liferay
- 'portal-ext.properties' is now a test resource rather than directly created via Ant 'echo'
- 'portal-ext.properties' now also activates Liferay developer mode, i.e. GoGo shell
  can now be used via Telnet
- read-me was updated to reflect current Liferay + Tomcat versions, a list of additionally
  needed JMX-related modules (for Arquillian artifact hot-deployment via JMX)
- read-me now also contains a trouble-shooting section explaining how to start
  JMX-related modules via GoGo shell in case they are not auto-activated after installation,
  which sporadically happens, especially when starting the newly installed Liferay server
  for the first time

Commit:
Bump Liferay Hot Deploy JMX Listener to 2.0.0-SNAPSHOT

It is ugly to use a snapshot (repository), but at least this version seems
to be compatible with LR 7.1 and does not need an additional copy of
com.liferay.portal:com.liferay.portal.kernel:2.0.0 with an older version
number. Probably this is what kep the installed modules from being
auto-activated in the first place.

Commit:
Fix forgotten dependency upgrades for arquillian-container-liferay

- LR portal kernel 3.36.0 (latest at time of commit)
- LR hot-deploy JMX listener 2.0.0-SNAPSHOT

csierra commented:
hi @kriegaex 
we have been very busy these days... sorry for the long delay. 
we will have a look as soon as we can.

Thx for you contribution!

Commit:
Add details to "known problems" section

Commit:
Update Liferay bundle path to 7.1 in travis.yml

Commit:
Add trailing slashes to folder names in .gitignore

This conforms to the description at
https://git-scm.com/docs/gitignore#_pattern_format.

Commit:
Fix diverging JUnit versions 4.11 vs. 4.12

Now the whole project uses 4.12 and the version is taken from property
'version.junit' in the root POM.

cgoncas commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
